src/client/Client.cc

```diff
@@ -4595,12 +4596,22 @@ int Client::check_permissions(Inode *in, int flags, int uid, int gid)
     }    
   }
 #endif
+  unsigned want = 0;
+  if ((flags & O_ACCMODE) == O_WRONLY)
+    want = 2;
+  else if ((flags & O_ACCMODE) == O_RDWR)
+    want = 6;
+  else if ((flags & O_ACCMODE) == O_RDONLY)
+    want = 4;
+
+  int ret = _posix_acl_permission(in, uid, gid, sgids, sgid_count, want);
+  if (ret != -EACCES)
```

Comment:
we just require one test pass. see kernel's acl check function

static int acl_permission_check(struct inode *inode, int mask)
{
        unsigned int mode = inode->i_mode;

        if (likely(uid_eq(current_fsuid(), inode->i_uid)))
                mode >>= 6;
        else {
                if (IS_POSIXACL(inode) && (mode & S_IRWXG)) {
                        int error = check_acl(inode, mask);
                        if (error != -EAGAIN)
                                return error;
                }

                if (in_group_p(inode->i_gid))
                        mode >>= 3;
        }

        /*
         * If the DACs are ok we don't need any capability check.
         */
        if ((mask & ~mode & (MAY_READ | MAY_WRITE | MAY_EXEC)) == 0)
                return 0;
        return -EACCES;
}